### PR TITLE
Display number of existing templates/(dataset,config) and add checkbox to filter priority datasets

### PR DESCRIPTION
**Pull request by manandey**
PR to display current number of templates created for each dataset/config

### CODE REVIEW

**Comment by stephenbach** [comment]
Awesome! I wonder if there's a way to display this info without loading the dataset? So as described in #3 , users could know what they should prioritize.

**Comment by shanyas10** [comment]
I was wondering if it would also be beneficial to sort the dataset_list in the (reversed) order of the number of templates present

**Comment by manandey** [comment]
> Awesome! I wonder if there's a way to display this info without loading the dataset? So as described in #3 , users could know what they should prioritize.

Sure! I will have a look at it.

**Comment by manandey** [comment]
Hi @stephenbach! The current PR is updated to accommodate #3 as well.

**Comment by shanyas10** [comment]
@stephenbach we were wondering if the number of templates should be displayed in the dropdown as well (since the current implementation would give us an idea of the number post-selection only)? Or we can have custom filters for datasets for filtering on basis of the number of existing templates ( in addition to others like language, as suggested in #22  )

File: templates.yaml [hunk 1]
```diff
@@ -4,44 +4,77 @@ ag_news:
...
     name: test1
+    output_fn: ''
+    prompt_fn: ''
     reference: example template_1
   basic: !Template
     jinja: "{{text}} \nIs this text an example of news about world politics, sports,\
       \ business, or technology? ||| \n{{[\"World politics\", \"Sport\", \"Business\"\
       , \"Technology\"][label] }}"
     name: basic
+    output_fn: ''
```
Comment (stephenbach) [inline templates.yaml:15]: Looks like this reverted to an old version of `templates.yaml`. There shouldn't be any *_fn fields anymore.

File: promptsource/templates.py [hunk 2]
```diff
@@ -67,7 +69,7 @@ def remove_template(self, dataset, template_name):
             raise ValueError(f"No template with name {template_name} " + f"for dataset {dataset} exists.")
 
         del self.templates[dataset][template_name]
-
+        self.templates[dataset]["count"] = num_templates
```
Comment (stephenbach) [inline promptsource/templates.py:72]: Is it necessary to maintain an explicit count of the templates? Would just calling `len(self.templates[dataset])` have the same effect and avoid the headache of having to keep track of the counts?

**Comment by stephenbach** [comment]
Thanks @manandey and @shanyas10 ! Left a couple comments.

Re: displaying number of templates in the dropdown, that would be nice! (Also need to make it work for subsets.)

**Review by manandey — COMMENTED** [review]
@stephenbach variable names updated in this.

File: templates.yaml (hunk 1 above)
Comment (manandey) [reply templates.yaml]: The changes have been updated!

**Review by stephenbach — APPROVED** [review]
LGTM!

**Pull request merged**